### PR TITLE
Fix native return typehint errors

### PR DESCRIPTION
**Pull request by shmax**
This fixes about half of the "Anonymous function should have native return typehint" phpstan errors. If this PR goes through with no problem I'll tackle the rest in a second PR.

### CODE REVIEW

File: src/Validator/Rules/ProvidedRequiredArguments.php [hunk 1]
```diff
@@ -42,10 +43,12 @@ public function getVisitor(ValidationContext $context)
...
                     }
+
+                    return null;
                 },
             ],
             NodeKind::DIRECTIVE => [
-                'leave' => static function (DirectiveNode $directiveNode) use ($context) {
+                'leave' => static function (DirectiveNode $directiveNode) use ($context) : ?VisitorOperation {
                     $directiveDef = $context->getDirective();
                     if (! $directiveDef) {
```
Comment (simPod) [inline src/Validator/Rules/ProvidedRequiredArguments.php:53]: I'd invert the condition

File: src/Validator/Rules/ProvidedRequiredArguments.php [hunk 2]
```diff
@@ -19,7 +20,7 @@ public function getVisitor(ValidationContext $context)
     {
         return [
             NodeKind::FIELD     => [
-                'leave' => static function (FieldNode $fieldNode) use ($context) {
+                'leave' => static function (FieldNode $fieldNode) use ($context) : ?VisitorOperation {
                     $fieldDef = $context->getFieldDef();
 
                     if (! $fieldDef) {
```
Comment (simPod) [inline src/Validator/Rules/ProvidedRequiredArguments.php:26]: I'd invert the condition

File: src/Validator/Rules/ProvidedRequiredArgumentsOnDirectives.php [hunk 3]
```diff
@@ -85,7 +85,7 @@ static function (NamedTypeNode $argument) : string {
         }
 
         return [
-            NodeKind::DIRECTIVE => static function (DirectiveNode $directiveNode) use ($requiredArgsMap, $context) {
+            NodeKind::DIRECTIVE => static function (DirectiveNode $directiveNode) use ($requiredArgsMap, $context) : ?string {
                 $directiveName = $directiveNode->name->value;
                 $requiredArgs  = $requiredArgsMap[$directiveName] ?? null;
                 if (! $requiredArgs) {
```
Comment (simPod) [inline src/Validator/Rules/ProvidedRequiredArgumentsOnDirectives.php:91]: I'd invert the condition

File: src/Validator/Rules/ValuesOfCorrectType.php [hunk 4]
```diff
@@ -77,6 +78,8 @@ public function getVisitor(ValidationContext $context)
 
                     return Visitor::skipNode();
                 }
+
+                return null;
```
Comment (simPod) [inline src/Validator/Rules/ValuesOfCorrectType.php:82]: I'd invert the condition so the `skipNode()` is in the main path and `null` as early exit.

File: phpstan-baseline.neon [hunk 5]
```diff
@@ -3,2230 +3,1690 @@ parameters:
 		-
 			message: "#^Only booleans are allowed in &&, GraphQL\\\\Language\\\\AST\\\\Node\\|null given on the left side\\.$#"
 			count: 1
-			path: src/Error/Error.php
+			path: src\Error\Error.php
```
Comment (simPod) [inline phpstan-baseline.neon:6]: Can you check why it regenerated backslashes? Does not seem correct 🤔Maybe bump phpstan if required.

File: src/Validator/Rules/ProvidedRequiredArguments.php (hunk 2 above)
Comment (shmax) [reply src/Validator/Rules/ProvidedRequiredArguments.php:26]: I don't disagree, but that sort of opens a can of worms and I think that kind of thing will be covered by other phpstan warnings which I'll tackle in a new PR; I'd rather focus on "return native typehint" errors in this PR if you don't mind.

File: src/Validator/Rules/ProvidedRequiredArgumentsOnDirectives.php (hunk 3 above)
Comment (shmax) [reply src/Validator/Rules/ProvidedRequiredArgumentsOnDirectives.php:91]: Will address in future PR.

File: src/Validator/Rules/ValuesOfCorrectType.php (hunk 4 above)
Comment (shmax) [reply src/Validator/Rules/ValuesOfCorrectType.php:82]: Will address in future PR.

File: phpstan-baseline.neon (hunk 5 above)
Comment (shmax) [reply phpstan-baseline.neon]: This was a flaw in phpstan innards, which was fixed in 0.12.18. I updated our lock to that version, made the necessary tweak to compose.json, and regenerated the file in 6bc5841. Thanks for catching that!

File: composer.json [hunk 6]
```diff
@@ -54,7 +54,7 @@
     "lint" : "phpcs",
     "fix" : "phpcbf",
     "stan": "phpstan analyse --ansi --memory-limit 2048M",
-    "phpstan-baseline": "phpstan analyse --ansi --error-format baselineNeon > phpstan-baseline.neon",
+    "phpstan-baseline": "phpstan analyse --ansi --generate-baseline=phpstan-baseline.neon",
```
Comment (simPod) [inline composer.json:57]:
```suggestion
    "phpstan-baseline": "phpstan analyse --ansi --generate-baseline",
```

File: src/Language/AST/NamedTypeNode.php [hunk 7]
```diff
@@ -9,6 +9,6 @@ class NamedTypeNode extends Node implements TypeNode
     /** @var string */
     public $kind = NodeKind::NAMED_TYPE;
 
-    /** @var NameNode */
+    /** @var string */
```
Comment (simPod) [inline src/Language/AST/NamedTypeNode.php:12]: Can you check if this is correct? There are some new errors in baseline I guess should not be there like

```
"#^Cannot access property \\$value on string\\.$#"
src/Validator/Rules/ProvidedRequiredArgumentsOnDirectives.php
```

File: composer.json (hunk 6 above)
Comment (shmax) [reply composer.json:57]: That works. d873515

File: src/Language/AST/NamedTypeNode.php (hunk 7 above)
Comment (shmax) [reply src/Language/AST/NamedTypeNode.php]: You're right (it can be either NameNode or string). Fixed in a702169

**Comment by shmax** [comment]
Thanks much for reviewing 👍

**Pull request merged**